lib/imapSimple.js

```diff
@@ -225,9 +261,49 @@ function connect(options, callback) {
     });
 }
 
+/**
+ * Returns a list of the parts.
+ * A part is either a message or an attachment.
+ * Parts are usually nested and numbered things like 1, 1.1, 1.2, ...
+ * Code taken from http://stackoverflow.com/questions/25247207/how-to-read-and-save-attachments-using-node-imap
+ */
+function findParts(struct, parts) {
+    parts = parts ||  [];
+    for (var i = 0; i < struct.length; i++) {
+        if (Array.isArray(struct[i])) {
+	    findParts(struct[i], parts);
+	} else if (struct[i].partID) {
+	    parts.push(struct[i]);
+	}
+    }
+    return parts;
+}
+
+/**
+ * Simplify the structure of the header (subject, date, etc) a bit
+ * Does some simple parsing to pull out useful attributes and all attachments
+ */
+function decorateMessage(message) {
+    if (message.parts.length != 1)
+        throw Error('Message has weird number of parts')
+    message._header = {};
+    ['from', 'to', 'subject', 'date'].forEach(function(key) {
```

Comment:
That's fine – it's not really needed for the attachment code anyway. Just `findParts`